src/main/java/org/sagebionetworks/bridge/spring/controllers/CRCController.java

```diff
@@ -315,41 +311,35 @@ String getObservationValue(Observation observation) {
-        IParser parser = FHIR_CONTEXT.newJsonParser();
-        String json = parser.encodeResourceToString(patient);
 
-        String cuimcEnv = (account.getDataGroups().contains(TEST_USER_GROUP)) ? "test" : "prod";
-        String cuimcUrl = "cuimc." + cuimcEnv + ".url";
-        String url = bridgeConfig.get(cuimcUrl);
-        url = resolveTemplate(url, ImmutableMap.of("patientId", account.getId()));
-
-        try {
-            HttpResponse response = put(url, json, account);
-            throwExceptions(response, account.getId());
-        } catch (IOException e) {
-            LOG.error(INT_ERROR + account.getId(), e);
-            throw new ServiceUnavailableException(UNAVAILABLE_ERROR);
-        }
-        LOG.info("Patient record submitted to CUIMC for user " + account.getId());
-    }
-*/
-    void addLocation(JsonNode node, Account account, String location) {
+    /**
+     * This is a nice-to-have addition of address information for the location given by an 
+     * ID in the appointment record. Do not fail the request if this fails, but log enough
+     * to troubleshoot if the issue is on our side.
+     */
+    void addLocation(JsonNode node, Account account, String locationId) {
         String cuimcEnv = (account.getDataGroups().contains(TEST_USER_GROUP)) ? "test" : "prod";
         String cuimcUrl = "cuimc." + cuimcEnv + ".location.url";
         String url = bridgeConfig.get(cuimcUrl);
-        url = resolveTemplate(url, ImmutableMap.of("location", location));
+        String reqBody = "id=\"" + locationId + "\"";
 
         try {
-            HttpResponse response = get(url, account);
-            throwExceptions(response, account.getId());
+            HttpResponse response = post(url, account, reqBody);
+            String resBody = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8.name());
             
-            String body = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8.name());
-            JsonNode locationJson = BridgeObjectMapper.get().readTree(body);
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode != 200 && statusCode != 201) {
+                LOG.warn("Error retrieving location, id = " + locationId + ", status = " + statusCode
+                        + ", response body = " + resBody);
+                return;
+            }
+            JsonNode bundleJson = BridgeObjectMapper.get().readTree(resBody);
+            if (!bundleJson.has("entry") || ((ArrayNode)bundleJson.get("entry")).size() == 0) {
+                LOG.warn("Error retrieving location, id = " + locationId + ", status = " + statusCode
+                        + ", response body = " + resBody);
+                return;
+            }
+            JsonNode locationJson = bundleJson.get("entry").get(0).get("resource");
             JsonNode telecom = locationJson.get("telecom");
             JsonNode address = locationJson.get("address");
```

Comment:
Should we also check that "resource" exists? Otherwise, we'd get a NullPointerException if they return JSON that doesn't contain "resource".

Reply:
It appears to be all or nothing. If there's an entry, its a resource.

Reply:
I can add NPE checks though, no harm.

Reply:
Updated